Add Caddy documentation and update category description

haseeeb07 opened the pull request.

haseeeb07 committed:
Add Caddy documentation and update category description

MatthewLeigh commented on docs/02-tools-and-technologies/04-caddy/02-science-island-configuration.md:
````diff
@@ -0,0 +1,134 @@
+### Local
+```bash
+cd si-auth-service
+docker compose up -d
````
`docker compose up -d --build` might be better for local dev, just to ensure local changes are applied.

MatthewLeigh commented on docs/02-tools-and-technologies/04-caddy/03-working-with-caddyfiles.md:
```diff
@@ -0,0 +1,161 @@
+
+This recreates the VM with the new config.
+
+## Adding a New Domain
```
```suggestion
## Adding a New Subdomain
```

MatthewLeigh commented on docs/02-tools-and-technologies/04-caddy/03-working-with-caddyfiles.md:
```diff
@@ -0,0 +1,161 @@
+
+### 2. Update DNS
+
+Add A record: `monitoring.scienceisland.com` → `<instance-ip>`
```
Might be good to add a brief comment here explaining how to go about this (e.g., contact Bill (CTO) to add an `A Record` with the domain provider for the new subdomain.) There's a good chance whoever is reading this hasn't deployed a website before, so they might need a bit of help here.

MatthewLeigh reviewed (commented):
All looks good! Just a couple of minor notes.

haseeeb07 committed:
docs: improve Caddy documentation based on feedback

- Add --build flag to docker compose command for better local dev experience
- Change 'Adding a New Domain' to 'Adding a New Subdomain' for accuracy
- Add guidance on DNS record updates (contact Bill CTO for domain provider changes)
